pyccel/ast/builtins.py

```diff
@@ -468,11 +478,14 @@ def __init__(self, *args):
                 self._rank  = len(self._shape)
 
         else:
-            self._rank      = max(a.rank for a in args) + 1
+            max_rank = max(a.rank for a in args)
+            self._rank      = max_rank + 1
             self._dtype     = NativeGeneric()
             self._precision = 0
             if self._rank == 1:
                 self._shape     = (LiteralInteger(len(args)), )
+            elif any(a.rank != max_rank for a in args):
+                self._shape     = (LiteralInteger(len(args)), ) + (None,)*(self._rank-1)
             else:
                 self._shape     = (LiteralInteger(len(args)), ) + args[0].shape
```

Comment:
It is not clear to me how these changes solve issue #1182. Can you please explain?

Reply:
Sorry I wrote too quickly. My question was about the neater error.

Reply:
So the problem was that we were trying to concatenate a tuple with `None`, correct?

Reply:
Yes exactly